### ISSUE TITLE
[Bug]: Textarea's label can't be accessed by tab and screen-reader

### ISSUE THREAD

**Issue by sherryzhao11** [microsoft/fluentui#30925]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 6.1 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (2) x64 AMD EPYC
    Memory: 1.95 GB / 4.01 GB
    Container: Yes
    Shell: Unknown
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codesandbox.io/p/devbox/px9kgf?migrateFrom=zt34dn&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clui1olug00063d5vtp8ryt9t%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clui1oluf00023d5vznlj218t%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clui1oluf00043d5vylpdgu0v%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clui1oluf00053d5vnhka58l6%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clui1oluf00023d5vznlj218t%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clui1oluf00013d5vnovdqolw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%257D%255D%252C%2522id%2522%253A%2522clui1oluf00023d5vznlj218t%2522%252C%2522activeTabId%2522%253A%2522clui1oluf00013d5vnovdqolw%2522%257D%252C%2522clui1oluf00053d5vnhka58l6%2522%253A%257B%2522id%2522%253A%2522clui1oluf00053d5vnhka58l6%2522%252C%2522activeTabId%2522%253A%2522clui1pb0e00ax3d5v9jgx82ee%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522Development%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clui1pb0e00ax3d5v9jgx82ee%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%255D%257D%252C%2522clui1oluf00043d5vylpdgu0v%2522%253A%257B%2522id%2522%253A%2522clui1oluf00043d5vylpdgu0v%2522%252C%2522activeTabId%2522%253A%2522clui1p764009v3d5v20004pkz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clui1oluf00033d5v28vqz51w%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clui1onje000nd8f80suy34c1%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522Development%2522%252C%2522id%2522%253A%2522clui1p764009v3d5v20004pkz%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Bug Description

For the component https://react.fluentui.dev/?path=/docs/components-textarea--default Textarea, the label can't be accessed by tab and screen-reader

Actual Results:
Label 'Small Textarea' is not read for Textarea.
Only placeholder is read.
Reader recording as https://github.com/microsoft/fluentui/assets/132635147/68bc48ac-ea80-49b0-a94a-d7fed8cdea94

Expected:
When the form field is accessed, first the visible label 'Small Textarea' should be read and then the placeholder should be read.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by spmonahan** [comment]
> the label can't be accessed by tab and screen-reader
Labels are not tabbable so this is expected behavior.

What screen reader, browser and OS are you testing with? I'm not able to repro the issue with Narrator in Edge on Windows 11.

**Comment by sherryzhao11** [comment]
In the Teams App on Win11  , and use JAWS to test reader.

**Comment by smhigley** [comment]
I'm not able to repro the issue on any of our docs site examples, using JAWS on Edge, Chrome, and Firefox.

I'd suggest checking that the label is correctly associated with the Textarea in the UI that is shown in the attached screen recording. If you are able to repro it on a Fluent control, please provide a link to the environment where you see the issue, along with information on the browser & JAWS version.

Thanks!
